### PR TITLE
Fixing off by one error in DatePickerRange.

### PR DESCRIPTION
**Pull request by ravij3**
1. Fix requires converting max_date_allowed and min_date_allowed into
moment.js date objects.
2. Added test to cover the scenario reported in [#867](https://github.com/plotly/dash-core-components/issues/867)

#### Tested locally with below script

```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : datepicker.py
# Author            : Ravi Jadhav 
# Date              : 19.02.2021
# Last Modified Date: 20.02.2021
# Last Modified By  : Ravi Jadhav
from datetime import date
import dash
import dash_html_components as html
import dash_core_components as dcc
import re

external_stylesheets = ['https://codepen.io/chriddyp/pen/bWLwgP.css']

app = dash.Dash(__name__, external_stylesheets=external_stylesheets)
app.layout = html.Div([
    dcc.DatePickerRange(
        id='my-date-picker-range',
        min_date_allowed=date(2021, 1, 10),
        max_date_allowed=date(2021, 1, 20),
        initial_visible_month=date(2021, 1, 10),
        end_date=date(2021, 1, 19),
        start_date=date(2021, 1, 11)
    ),
    html.Div(id='output-container-date-picker-range')
])


@app.callback(
    dash.dependencies.Output('output-container-date-picker-range', 'children'),
    [dash.dependencies.Input('my-date-picker-range', 'start_date'),
     dash.dependencies.Input('my-date-picker-range', 'end_date')])
def update_output(start_date, end_date):
    string_prefix = 'You have selected: '
    if start_date is not None:
        start_date_object = date.fromisoformat(start_date)
        start_date_string = start_date_object.strftime('%B %d, %Y')
        string_prefix = string_prefix + 'Start Date: ' + start_date_string + ' | '
    if end_date is not None:
        end_date_object = date.fromisoformat(end_date)
        end_date_string = end_date_object.strftime('%B %d, %Y')
        string_prefix = string_prefix + 'End Date: ' + end_date_string
    if len(string_prefix) == len('You have selected: '):
        return 'Select a date to see it displayed here'
    else:
        return string_prefix


if __name__ == '__main__':
    app.run_server(debug=True)
```

Originally you cant select min and max date as they are off by one. In the below screenshot we can see that both Jan 10th and Jan 20th are selectable.

#### Before selection

image

#### After Selection
image

### CODE REVIEW

**Review by alexcjohnson — APPROVED** [review]
Nice work @ravij3 - exactly the right fix and a very nice test! 💃 

We're having some issues with our visual CI tests (Percy) right now but once we sort those out this should be ready to merge!

**Comment by ravij3** [comment]
Thanks @alexcjohnson !! I will pick up some other fixes in the meantime.

File: tests/integration/calendar/test_date_picker_range.py [hunk 1]
```diff
@@ -80,3 +80,31 @@ def test_dtpr003_no_initial_month_no_min_date_start_date(dash_dcc):
...
+                max_date_allowed=datetime(2021, 1, 20),
+                min_date_allowed=datetime(2021, 1, 10),
+            )
+        ]
+    )
+
+    dash_dcc.start_server(app)
+
+    dash_dcc.select_date_range('dps-initial-month',(10,20))
+
```
Comment (alexcjohnson) [inline tests/integration/calendar/test_date_picker_range.py:101]:
```suggestion
    dash_dcc.select_date_range('dps-initial-month', (10, 20))

```

File: tests/integration/calendar/test_date_picker_range.py [hunk 2]
```diff
@@ -80,3 +80,31 @@ def test_dtpr003_no_initial_month_no_min_date_start_date(dash_dcc):
...
     )
+
+def test_dtpr004_max_and_min_dates_are_clickable(dash_dcc):
+    app = dash.Dash(__name__)
+    app.layout = html.Div(
+        [
+            dcc.DatePickerRange(
+                id="dps-initial-month",
+                start_date=datetime(2021, 1, 11),
+                end_date=datetime(2021,1,19),
```
Comment (alexcjohnson) [inline tests/integration/calendar/test_date_picker_range.py:91]:
```suggestion
                end_date=datetime(2021, 1, 19),
```

File: tests/integration/calendar/test_date_picker_range.py [hunk 3]
```diff
@@ -80,3 +80,31 @@ def test_dtpr003_no_initial_month_no_min_date_start_date(dash_dcc):
         "#dps-initial-month .CalendarMonth.CalendarMonth_1[data-visible=true] strong",
         "August 2019",
     )
+
+def test_dtpr004_max_and_min_dates_are_clickable(dash_dcc):
```
Comment (alexcjohnson) [inline tests/integration/calendar/test_date_picker_range.py:84]:
```suggestion

def test_dtpr004_max_and_min_dates_are_clickable(dash_dcc):
```

**Pull request merged**